kategory/src/main/kotlin/kategory/data/SetKW.kt

```diff
@@ -0,0 +1,31 @@
+package kategory
+
+@higherkind
+@deriving(Foldable::class, MonoidK::class)
+data class SetKW<A>(val set: Set<A>) : SetKWKind<A>, Set<A> by set {
+
+    fun <B> foldL(b: B, f: (B, A) -> B): B = this.fold(b, f)
+
+    fun <B> foldR(lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: SetKW<A>): Eval<B> = when {
+            fa_p.set.isEmpty() -> lb
+            else -> f(fa_p.set.first(), Eval.defer { loop(fa_p.set.drop(1).toSet().k()) })
```

Comment:
`SetKW.foldR` has the same problem like `ListKW.foldR` related in #240, seems it's a problem of Eval #252